src/gkmodel/node_model.py

```diff
@@ -232,17 +236,32 @@ def predict(self,
             for study_id in data.study_id
         ])
 
+        adjust_values = np.zeros(self.data.num_points)
+
         if slope_quantile is not None:
-            for name, quantile in slope_quantile.items():
-                if name in self.cov_names:
-                    i = self.cov_names.index(name)
-                    v = np.quantile(soln[:, i], quantile)
-                    if quantile >= 0.5:
-                        soln[:, i] = np.maximum(soln[:, i], v)
-                    else:
-                        soln[:, i] = np.minimum(soln[:, i], v)
-
-        return np.sum(mat*soln, axis=1)
+            covs_index = [self.cov_names.index(name) for name, _ in slope_quantile.items() if name in self.cov_names]
+            quantiles = [quantile for name, quantile in slope_quantile.items() if name in self.cov_names]
+            
+            if ref_cov is not None:
+                ref_mat = deepcopy(mat)
+                for study in self.data.studies:
+                    study_index = self.data.study_id == study
+                    ref_index = study_index & (self.data.covs[ref_cov[0]] == ref_cov[1])
```

Comment:
here I don't know if want to make sure `ref_index` only have one `True`, or we could take the first `True`.

Reply:
I added a check forcing it only to have one `True`.